# L19 worksheet

## Printing and comments

In this section, you will learn how to:

- Use the `print()` function to display information.
- Write comments to explain your code.

**Related material:** [printing](https://python.crumblearn.org/basics/printing/) and [comments](https://python.crumblearn.org/basics/comments/)

The `print()` function is used to display output on the screen.
Here's a simple example:

In [ ]:
print("Computational biology is the best major!")

In [ ]:
print("Welcome to Python programming!")

In [ ]:
print("Welcome", "to", "Python", "programming!")

In [ ]:
print(42)
print(3.14)

In [ ]:
print("The answer is:", 42)

In [ ]:
print("Sometimes I want to\n\n\nbe dramatic.")

Comments are used to explain what your code does.
They are ignored by Python when the code is run.

In [ ]:
# This is a single-line comment
print("Comments are ignored by Python.")

In [ ]:
"""
This is a multi-line comment.
It can span multiple lines.
"""

print("Use comments to explain your code!")

**🏅 Exercises**

**1.1** In the cell below, print your name, age, and favorite color on one line.
For example

```text
Alex, 22, Purple
```

**1.2** In the cell below, print the same information but only one on each line.
For example,

```text
Alex
22
Purple
```

**1.3** In the cell below, print the same text as before but using only one `print` statement.

## Arithmetic operations

In this section, you will learn how to:

- Perform basic arithmetic operations using Python.
- Understand operator precedence.

**Relevant material:** [Arithmetic](https://python.crumblearn.org/basics/arithmetic/)

Python supports the following arithmetic operators:

| **Operator** | **Description**             | **Example**             |
|--------------|-----------------------------|-------------------------|
| `+`          | Addition                    | `2 + 3` → `5`           |
| `-`          | Subtraction                 | `7 - 4` → `3`           |
| `*`          | Multiplication              | `3 * 4` → `12`          |
| `/`          | Division                    | `8 / 2` → `4.0`         |
| `//`         | Floor Division              | `7 // 3` → `2`          |
| `%`          | Modulus (remainder)         | `7 % 3` → `1`           |
| `**`         | Exponentiation              | `2 ** 3` → `8`          |

In [ ]:
# Addition
print(5 + 3)

In [ ]:
# Subtraction
print(10 - 7)

In [ ]:
# Multiplication
print(4 * 6)

In [ ]:
# Division
print(9 / 3)

In [ ]:
# Floor Division
print(7 // 2)

In [ ]:
# Modulus
print(8 % 3)

In [ ]:
# Exponentiation
print(2**4)

When multiple operators are used in a single expression, Python follows **operator precedence** (order of operations):

1. Parentheses `()`
2. Exponentiation `**`
3. Multiplication `*`, Division `/`, Floor Division `//`, Modulus `%`
4. Addition `+`, Subtraction `-`

In [ ]:
# Parentheses first
print((2 + 3) * 4)

In [ ]:
# Exponentiation before multiplication
print(2**3 * 4)

In [ ]:
# Left-to-right for operators with same precedence
print(10 / 2 * 3)

**🏅 Exercises**

**2.1** Evaluate the expression $\frac{(3 + 5)^{2}}{2 \cdot 4}$.
(Hint: the answer is `8.0`.)

## Variables

In this section, you will learn:

- How to define and use variables in Python.
- Rules for naming variables.
- The importance of variables in programming.

**Relevant material:** [variables](https://python.crumblearn.org/basics/variables/)

Variables are containers for storing data values.
Think of them as labels for information you want to use in your program.

In [4]:
# Defining variables
name = "Alice"
age = 25
pi = 3.141592

# Using variables
print(name)
print(age)
print(pi)

Alice
25
3.141592


In [ ]:
name, age, pi = "Bob", 111, 3.141_592
print(name, age, pi)

name = "Bill"
print(name, age, pi)

Bob 111 3.141592
bill 111 3.141592


Python automatically assigns a type to a variable based on the value you assign.

| **Type**    | **Description**          | **Example**     |
|-------------|--------------------------|-----------------|
| `int`       | Integer (whole number)   | `age = 25`      |
| `float`     | Decimal number           | `pi = 3.14`     |
| `str`       | String (text)            | `name = "Bob"`  |
| `bool`      | Boolean (True/False)     | `is_valid = True`|

In [ ]:
x = 10
print(type(x))

y = 3.14
print(type(y))

z = "Hello"
print(type(z))

**🏅 Exercises**

**3.1**: Using the variables `length`, `height`, and `width`, compute the volume of a box with dimensions 10, 5, and 3, respectively.

**3.2**: Imagine drawing a line from one corner of the rectangular box to the opposite corner, creating a right triangle with sides representing the length, width, and height; the diagonal is the hypotenuse of this triangle with length $\sqrt{l^2 + h^2 + w^2}$.
Calculate this diagonal length using the same `length`, `height`, and `width` variables below.

**3.3**: Change the `length`, `height`, and `width` variables defined in **3.1** to be 8.32, 5.433, and 12.1 and recompute the diagonal using the same cell in **3.2**.